src/tls13handshake.rs

```diff
@@ -298,10 +298,22 @@ fn put_server_signature(
         let cert = parse_server_certificate(&algs, sc)?;
         let tx = transcript_add1(tx, sc);
         let th_sc = get_transcript_hash(&tx)?;
-        let pk = verification_key_from_cert(&cert)?;
+        let (signature_scheme, certificate_key) = verification_key_from_cert(&cert)?;
         let sig = parse_certificate_verify(&algs, scv)?;
         let sigval = PREFIX_SERVER_SIGNATURE.concat(&th_sc);
-        verify(&sig_alg(&algs), &pk, &bytes(&sigval), &sig)?;
+        // TODO(Refactor): This is used because the API of `evercrypt_cryptolib::verify` wants a `PublicVerificationKey`.
+        //                 That enum is directly unpacked in `verify`, though.
+        if signature_scheme == SignatureScheme::EcdsaSecp256r1Sha256 {
+            let verification_key = ecdsa_public_key(&cert, certificate_key)?;
+            let pk = PublicVerificationKey::EcDsa(verification_key);
+            verify(&sig_alg(&algs), &pk, &bytes(&sigval), &sig)?;
+        } else if signature_scheme == SignatureScheme::RsaPssRsaSha256 {
```

Comment:
Neither do we have `else if` or inline blocks. See https://github.com/hacspec/hacspec/issues/85 for the `else if` issue.